src/compression/serialize.rs

```diff
@@ -0,0 +1,247 @@
+use std::io;
```

Comment:
I might be able to import some utility functions from `clvm_rs` rather than pasting them here. A separation in `clvm_rs::serialize` into two files (one with serialization and the other with deserialization functions) might be helpful here.